CSVSource/README.md

```diff
@@ -0,0 +1,213 @@
+"field1"="f" or "field3"="f" . 
+
+### Schema Configuration
+Schema can be used to control the field names on the uploaded event. If no name is assigned, 'fieldX' will be used where 'X' is the index of the field in the line.  For example field0, field1, etc. 
+
+Those values can be used to override that default behavior for example, the following 
+
+will cause the attribute of the 3rd value in each line to be called "address". 
+So instead of loading event : 
+
+`{ field0:1, field1:true, field2:"there"}` it will upload `{ field0:1, field1:true, address:"there"}`
+this can save conversion processing on the server.
+
+### Execution Options
+
+*   **maxActiveTasks**: Optional. The maximum number of threads running at any given point. This is the number of CSV files being processed simultaneously. Must be a positive integer. Default value is 5.
+*   **maxQueuedTasks**: Optional. The maximum number of queued tasks at any given point for CSV files,overflowing that number might cause lack of data . Must be a positive integer. Default value is 10.
+*   **processExistingFiles**: Optional. If set to `true`, the service will process all files already existing in the folder `fileFolderPath` (filtered using `filePrefix` and the `fileExtension`). Otherwise the service will process only new files.  Default is`false`.
+*   **extensionAfterProcessing**: Optional. Rename the file after it has been processed to avoid reprocessing (_e.g._ for cases where `processExistingFiles` is set to `true`).  The default value is combination of the 'fileExtension' and `done`.  For example `.csv.done` when `fileExtension` set to `.csv`.
+*   **deleteAfterProcessing**: Optional. Delete the processed file only if processed successfully to avoid reprocessing in cases where `processExistingFiles` is set to `true`. Default value is `false`
+
+## Messages from the Source
+
+Messages that are sent to the source as Notifications are JSON objects in the following format:
+```
+{
+   "file": "d:/tmp/csv/ejesmall.csv",
+   "segment": 39,
+   "lines": [
+      {
+         "flag": "1",
+         "value": "109411211",
+         "YScale": "13.8000"
+      },
+      {
+         "flag": "1",
+         "value": "109415211",
+         "YScale": "112.3507"
+      },
+      {
+         "flag": "1",
+         "value": "109416211",
+         "YScale": "-111.5000"
+      },
+      {
+         "flag": "1",
+         "value": "109432211",
+         "YScale": "1.0440"
+      },
+      {
+         "flag": "1",
+         "value": "109433211",
+         "YScale": "-0.3000"
+      }
+   ]
+}
+```
+number for lines will not exeeded the value determined by the `maxLinesInEvent` configuratipon parameter .
+
+## Event structure
+Each event consists on the follwoing structure :
+
+* **file** : the source file from where the data was extracted
+* **segment** : number of CSV lines on the current segment 
+* **lines** : the json buffer itself . where the key values are used from the schema definition . 
+
```

Comment:
Here, there should be an example (which should be added to the app in your CSVExample.zip) that shows how to process an incoming event.  it doesn't need to be anything fancy, just showing that the message from the source comes in, that the values can be seen, and that the data can be dealt with -- stuffed into a database record, logged, whatever.  This is the only documentation there is -- if you add the structure here, customers can download & try things without having to invent from scratch.

Reply:
done .